src/components/button/button.tsx

```diff
@@ -2,150 +2,123 @@ import type { ReactNode } from 'react';
 import { DEFAULT_OUTLINE_OFFSET, DEFAULT_TRANSIITON, fontWeight } from '../../styles';
 import type { ColorType } from '../../styles';
 import styled, { css } from 'styled-components';
-import moonbeam from '../../assets/moonbeam.svg';
-import metamask from '../../assets/metamask.svg';
 import { DEFAULT_BORDER_RADIUS, pxToRem, MAIN_MAX_WIDTH } from '../../styles';
 import { isLightTheme, useStore } from '../../helpers';
 import { Spinner } from '../../components';
 
-const icons = {
-	moonbeam,
-	metamask
-};
-
 interface CommonProps {
+	children: ReactNode;
 	disabled?: boolean;
-	children?: ReactNode;
 	isLoading?: boolean;
 	onClick: () => void;
 }
 
 interface PrimaryProps {
 	variant?: 'primary';
-	color: 'default' | 'transparent';
-	icon?: never;
+	color?: 'default' | 'transparent';
 }
 
 interface PureProps {
-	variant?: 'pure';
-	color?: never;
-	icon?: never;
+	variant: 'pure';
+	color?: 'default';
 }
 
-export type IconOnlyProps = {
-	icon?: 'moonbeam' | 'metamask';
-	color?: never;
-};
-
-interface ColorOnlyProps {
-	color?: ColorType;
-	icon?: never;
-}
-
-type ColorIconProps = IconOnlyProps | ColorOnlyProps;
-
 type SecondaryProps = {
 	variant: 'secondary';
-} & ColorIconProps;
+} & { color?: ColorType };
 
 type IndividualProps = PrimaryProps | SecondaryProps | PureProps;
 export type ButtonProps = IndividualProps & CommonProps;
 
-const StyledButton = styled.button(
-	({ variant = 'primary', color = 'default', disabled = false, icon, isLoading }: ButtonProps) => {
-		const isPrimary = variant === 'primary';
-		const isSecondary = variant === 'secondary';
-		const isPure = variant === 'pure';
-		const setColor = icon ? 'icon' : color;
-		const isColorDefault = setColor === 'default';
-		const isPrimaryTransparent = variant === 'primary' && color === 'transparent';
-		const isSecondaryDefault = isSecondary && setColor === 'default';
-		const {
-			state: { theme }
-		} = useStore();
-
-		return css`
-			display: ${icon || isLoading ? 'inline-flex' : 'inline-block'};
-			align-items: center;
-			justify-content: ${isLoading ? 'center' : 'space-between'};
-			max-width: ${isPrimary ? MAIN_MAX_WIDTH : pxToRem(160)};
-			width: 100%;
-			font-weight: ${isPrimary ? fontWeight.semibold : fontWeight.regular};
-			cursor: ${disabled ? 'not-allowed' : 'pointer'};
-			font-size: ${isPrimary ? pxToRem(16) : pxToRem(14)};
-			min-height: ${isPrimary ? pxToRem(57) : pxToRem(35)};
-			padding: ${pxToRem(4)} ${pxToRem(12)};
-			color: ${isPure
-				? theme.font.default
-				: isSecondaryDefault || isPrimaryTransparent
+const StyledButton = styled.button((props: ButtonProps) => {
```

Comment:
I checked out the `styled-components` docs sine we are randomly using either `props` or the destructed object and it seems that using `props` is the recommended way in `styled-components`. That's why I changed the structure and since we have default values defined for the component itself we don't really need them here, correct. But they are still there (coming from the top level - the component itself).